scripts/in_container/run_fix_ownership.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python3
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+import os
+import pwd
+import sys
+from pathlib import Path
+
+HOST_OS = os.environ.get("HOST_OS", "")
+DOCKER_IS_ROOTLESS = os.environ.get("DOCKER_IS_ROOTLESS", "false") == "true"
+
+
+def change_ownership_of_files(path: Path) -> None:
+    host_user_id = os.environ.get("HOST_USER_ID", "")
+    host_group_id = os.environ.get("HOST_GROUP_ID", "")
+    if host_user_id == "" or host_group_id == "":
+        print(
+            f"ERROR: HOST_USER_ID or HOST_GROUP_ID environment variables "
+            f"are not set: {host_user_id}:{host_group_id}"
+        )
+        sys.exit(1)
+    if not host_user_id.isnumeric() or not host_group_id.isnumeric():
+        print(
+            f"ERROR: HOST_USER_ID or HOST_GROUP_ID environment variables "
+            f"should be numeric: {host_user_id}:{host_group_id}"
+        )
+        sys.exit(1)
+    count_files = 0
+    root_uid = pwd.getpwnam("root").pw_uid
+    print("Attempting to see if there are files that need to be changed to host user ownership")
+    for file in path.rglob("*"):
+        try:
+            if file.is_symlink() and file.lstat().st_uid == root_uid:
+                # Also change ownership of symlink itself (by default it will only change
+                # ownership of the file it points to)
+                os.chown(file, int(host_user_id), int(host_group_id), follow_symlinks=False)
+                count_files += 1
+            if file.stat().st_uid == root_uid:
+                os.chown(file, int(host_user_id), int(host_group_id))
+                count_files += 1
+                if os.environ.get("VERBOSE_COMMANDS", "false") == "true":
+                    print(f"Changed ownership of {file}")
```

Comment:
```suggestion
            follow_symlinks = True
            if file.is_symlink() and file.lstat().st_uid == root_uid:
                follow_symlinks = False
           
           os.chown(file, int(host_user_id), int(host_group_id), follow_symlinks=follow_symlinks)
           count_files += 1
           if os.environ.get("VERBOSE_COMMANDS", "false") == "true":
                    print(f"Changed ownership of {file}")
```

Reply:
minor suggestion. :)

Reply:
Nope. It's good we want to change ownership for both - symlink and file it points at. The first if changes the ownership for the symlink (when it is one). The next if changes the ownership for either the file (if it is the file) or for the file symlink points at (if it's a symlink) by default `stat()` and `chown` follow symlinks so they only change the ownership of target file.

Reply:
I updated comments to explain it a bit better.

Reply:
Sorry...I just assumed it was a if-else block. My bad.

Reply:
Well. It was misleading - that's why it's good you noticed it - now I think the comment is more accurate - because it is quite surprising behviour (it was certainly quite surprising for me when I tested it today and notice a bunch of symlinks that continue to have root ownership even if I thought I changed them all :D

Reply:
Ya, it is surprising now that it's pointed out. I wasn't aware of it either. :)